panic: import "golang.org/x/net/html" error: cannot use type *golang.org/x/net/html.parser as type

Hi,

The following program sample.go triggers an unexpected result.
Error while trying to import "golang.org/x/net/html" 
Using Linux, Ubuntu 20.04.2 LTS

```
package main
import (
	"github.com/traefik/yaegi/interp"
	"github.com/traefik/yaegi/stdlib"
)

const src = `package foo
import (
	"golang.org/x/net/html"
)`

func main() {
	i := interp.New(interp.Options{
		GoPath: "/home/yaniv/go",
	})
	i.Use(stdlib.Symbols)
	_, err := i.Eval(src)
	if err != nil {
		panic(err)
	}
}
```

### Expected Result:

``No Error``

### Got:

```
panic: 3:2: import "golang.org/x/net/html" error: /home/yaniv/go/src/golang.org/x/net/html/parse.go:2201:17: cannot use type *golang.org/x/net/html.parser as type 

goroutine 1 [running]:
main.main()
	/home/yaniv/go/src/test-yagi/sample.go:21 +0xdb
Process exiting with code: 2 signal: false

```

Thank you

--- a/interp/interp.go
+++ b/interp/interp.go
@@ -770,7 +770,7 @@ func (interp *Interpreter) ImportUsed() {
 			}
 			name = key2name(fixKey(k))
 		}
-		sc.sym[name] = &symbol{kind: pkgSym, typ: &itype{cat: binPkgT, path: path.Dir(k), scope: sc}}
+		sc.sym[name] = &symbol{kind: pkgSym, typ: &itype{cat: binPkgT, path: k, scope: sc}}
 	}
 }
 

--- a/interp/type.go
+++ b/interp/type.go
@@ -877,6 +877,10 @@ func (t *itype) isRecursive() bool {
 	return false
 }
 
+func (t *itype) isIndirectRecursive() bool {
+	return t.isRecursive() || t.val != nil && t.val.isIndirectRecursive()
+}
+
 // isVariadic returns true if the function type is variadic.
 // If the type is not a function or is not variadic, it will
 // return false.

--- a/interp/typecheck.go
+++ b/interp/typecheck.go
@@ -51,7 +51,7 @@ func (check typecheck) assignment(n *node, typ *itype, context string) error {
 		return nil
 	}
 
-	if typ.isRecursive() || typ.val != nil && typ.val.isRecursive() {
+	if typ.isIndirectRecursive() || n.typ.isIndirectRecursive() {
 		return nil
 	}
 
@@ -997,6 +997,8 @@ func getArg(ftyp *itype, i int) *itype {
 		return arg
 	case i < l:
 		return ftyp.in(i)
+	case ftyp.cat == valueT && i < ftyp.rtype.NumIn():
+		return &itype{cat: valueT, rtype: ftyp.rtype.In(i)}
 	default:
 		return nil
 	}

